File: javascript-sdk/src/common/hash.ts
const encoder = new TextEncoder();
const hash = "SHA-256";
const name = "HMAC";

export async function sha256Hash(message: Uint8Array<ArrayBuffer>) {
  return await crypto.subtle.digest(hash, message);
}

export async function hmacSha256(
  key: Uint8Array<ArrayBuffer>,
  message: Uint8Array<ArrayBuffer>,
) {
  const hashKey = await crypto.subtle.importKey(
    "raw",
    key,
    { name, hash },
    false,
    ["sign"],
  );
  return await crypto.subtle.sign(name, hashKey, message);
}

File: dart-sdk/lib/src/common/hash.dart
import 'dart:convert';
import 'dart:typed_data';

import 'package:crypto/crypto.dart';

Digest sha256Hash(Uint8List message) {
  return _calculateHash(sha256, message);
}

Digest hmacSha256(Uint8List key, Uint8List message) {
  final h = Hmac(sha256, key);
  return _calculateHash(h, message);
}

Digest _calculateHash(Converter<List<int>, Digest> h, Uint8List data) {
  return h.convert(data);
}
